cmd/xl-v1-healing-common.go

```diff
@@ -97,11 +101,50 @@ func listOnlineDisks(disks []StorageAPI, partsMetadata []xlMetaV1, errs []error)
 }
 
 // outDatedDisks - return disks which don't have the latest object (i.e xl.json).
-func outDatedDisks(disks, latestDisks []StorageAPI, partsMetadata []xlMetaV1, errs []error) (outDatedDisks []StorageAPI) {
+func outDatedDisks(disks, latestDisks []StorageAPI, partsMetadata []xlMetaV1,
+	bucket, object string) (outDatedDisks []StorageAPI) {
+
 	outDatedDisks = make([]StorageAPI, len(disks))
 	for index, disk := range latestDisks {
+		// disk either has a older xl.json or is not online.
 		if disk == nil {
 			outDatedDisks[index] = disks[index]
+			continue
+		}
+
+		// disk has a valid xl.json but may not have all the
+		// parts. This is considered an outdated disk, since
+		// it needs healing too.
+		for pIndex, part := range partsMetadata[index].Parts {
+			if _, sErr := disk.StatFile(bucket,
```

Comment:
Are you suggesting that hashSum would return `errFileNotFound` in the case of missing parts? Yes, that makes sense.